client/src/components/Settings/SettingsCanvases.res

```diff
@@ -24,10 +28,30 @@ let setInfo = (
   orgs: list<string>,
   orgCanvasList: list<string>,
 ): t => {
+  ..._state,
   canvasList: canvasList,
   username: username,
   orgs: orgs,
   orgCanvasList: orgCanvasList,
 }
 
-let update = (state: t, ()) => state
+let validateCanvasName = (newCanvasName: Utils.formField): Utils.formField =>{
+  let error = {
+    let canvasName = newCanvasName.value
+    if !Js.Re.test_(%re("/^[A-Za-z0-9-]*$/"), canvasName){
+      Some("Invalid Canvas Name")
+    }else{
+        None
+    }
+  }
+  {...newCanvasName, error: error}
+}
+
+
+let update = (state: t, msg: msg):(t) =>
+  switch msg {
+    | Update(value) =>
+      let isInvalid= validateCanvasName(state.newCanvasName)
```

Comment:
here

Reply:
Thanks! it is working properly now.

![dark-Dark-Google-Chrome-2022-11 (2)](https://user-images.githubusercontent.com/87861924/203163286-b6bd733e-588e-4cff-923e-41f4d6354e3e.gif)